src/mongocxx/private/database.hpp

```diff
@@ -34,6 +34,19 @@ class database::impl {
         name(std::move(name))
     {}
 
+    impl(const impl& i) : 
+        database_t{libmongoc::database_copy(i.database_t)},
+        client_impl{i.client_impl},
+        name{i.name} 
+    {}
+
+    impl& operator=(const impl& i) {
+        database_t = libmongoc::database_copy(i.database_t);
```

Comment:
I think we raced to comment but I think we should explicitly delete it if we aren't going to be using it. In either case I don't really feel that strongly about it, as both ways are correct, i just prefer the way that results in less code.